### ISSUE TITLE
Feature: implement HiveSource to integrate with new processor

### ISSUE THREAD

**Issue by sundy-li** [datafuselabs/databend#6588]
**Summary**

Description for this feature.

After #5895  `HiveSource` is a dummy implementation, so we must turn off `enable_new_processor_framework` to query hive data.
It's better to make `HiveTable` read support the new processor.

**Comment by BohuTANG** [comment]
FYI @sandflee

**Comment by sandflee** [comment]
plan to  support the new processor in last two weeks.  cc @dantengsky

**Comment by zhang2014** [comment]
new processor HiveSource is implemented in #6584

**Comment by sandflee** [comment]
@zhang2014  thanks for your work, do you mind to move HiveSource related code to a new PR,  we could test and discuss in new  PR,  #6584 is too large.

**Comment by sundy-li** [comment]
> @zhang2014 thanks for your work, do you mind to move HiveSource related code to a new PR, we could test and discuss in new PR, #6584 is too large.

Agreed.  And I think we can apply projection pushdown into `HiveSource` after we have a separated pr.

**Comment by BohuTANG** [comment]
> > @zhang2014 thanks for your work, do you mind to move HiveSource related code to a new PR, we could test and discuss in new PR, #6584 is too large.
> 
> Agreed. And I think we can apply projection pushdown into `HiveSource` after we have a separated pr.

I think we can wait for @zhang2014 to finish his migration and then start the next work.

**Comment by zhang2014** [comment]
> > > @zhang2014 thanks for your work, do you mind to move HiveSource related code to a new PR, we could test and discuss in new PR, #6584 is too large.
> > 
> > 
> > Agreed. And I think we can apply projection pushdown into `HiveSource` after we have a separated pr.
> 
> I think we can wait for @zhang2014 to finish his migration and then start the next work.

It's ready for review.
